pkg/elevate/elevate.go

```diff
@@ -0,0 +1,90 @@
+package elevate
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+
+	"github.com/openshift/backplane-cli/pkg/utils"
+	logger "github.com/sirupsen/logrus"
+	"k8s.io/client-go/tools/clientcmd/api"
+)
+
+var (
+	OsRemove              = os.Remove
+	ExecCmd               = exec.Command
+	ReadKubeConfigRaw     = utils.ReadKubeconfigRaw
+	WriteKubeconfigToFile = utils.CreateTempKubeConfig
```

Comment:
I guess above 1 to  1 function mapping is needed since the unit test can override the function variable and returns dummy data.
You can achieve the same result by moving this to an interface and mocking the interface in test classes.  As an example 
```
# step 1
type ElevateCommand interface{
         OsRemove(name string) error
         exec.Command() .. 
         ReadKubeConfigRaw() *api.config
         WriteKubeconfigToFile(config *api.config)
}

# step 2
# generate mock classes via `mockgen -source=cmd/ocm-backplane/pkg/elevate/elevate.go cmd/ocm-backplane/pkg/elevate/mock` 

# step 3
# Use the mock function in unit tests 

```
I'm not strongly opposed to the current way of testing but a different way to create mocks.

Reply:
Thanks for the review @samanthajayasinghe  - I'm kind of against using mocks unless I really have to, as I feel they can create coupling between methods and make testing more difficult - but that's just me. Especially with system methods, and methods that aren't a part of the package itself (like `os.Remove`/`exec.Command`). I definitely think mocks have a place for things that you have written yourself, though.

Reply:
+1 I'm against using interfaces to extend system methods